### ISSUE TITLE
Rule: Warn for unguarded for-in loops

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#1]
Legacy from JSHint/JSLint: warn when you do this:

``` js
for (prop in obj) {
    // body
}
```

Instead of this:

``` js
for (prop in obj) {
    if (obj.hasOwnProperty(prop)) {

    }
}
```

Must also work even when the `for-in` loop body doesn't have braces around it (aka, whether or not there is a BlockStatement as the body).

Rule name should be `guard-for-in`
